### PR TITLE
[Review Needed]Pytest Style core/test_atomgroup.py

### PR DESCRIPTION
**Pull request by utkbansal**
Fixes #

Changes made in this Pull Request:
 - 

PR Checklist
------------
 - [ ] Tests?
 - [ ] Docs?
 - [ ] CHANGELOG updated?
 - [ ] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/core/test_atomgroup.py [hunk 1]
```diff
@@ -908,295 +905,303 @@ def test_instantselection_termini():
...
-                           self.universe.atoms.ix[:4])
-
-    def test_getitem_slice2(self):
-        assert_equal(self.universe.atoms[0:8:2].ix,
-                     self.universe.atoms.ix[0:8:2])
-
-    def test_getitem_str(self):
-        ag1 = self.universe.atoms['HT1']
+    dih_prec = 2
+    # def setUp(self):
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/core/test_atomgroup.py]: del

File: testsuite/MDAnalysisTests/core/test_atomgroup.py [hunk 2]
```diff
@@ -503,86 +497,87 @@ def test_split_residues(self):
...
-    def test_wrap_fragments(self):
-        ag = self.u.atoms[:250]
+    def test_wrap_fragments(self, u):
+        ag = u.atoms[:250]
         ag.wrap(compound='fragments')
 
         cen = np.vstack([f.center_of_mass() for f in ag.fragments])
 
-        assert_equal(self._in_box(cen), True)
+        assert_equal(self._in_box(cen, u), True)
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/core/test_atomgroup.py]: just use `assert`

File: testsuite/MDAnalysisTests/core/test_atomgroup.py [hunk 3]
```diff
@@ -503,86 +497,87 @@ def test_split_residues(self):
...
-    def test_wrap_segments(self):
-        ag = self.u.atoms[1000:1200]
+    def test_wrap_segments(self, u):
+        ag = u.atoms[1000:1200]
         ag.wrap(compound='segments')
 
         cen = np.vstack([s.atoms.center_of_mass() for s in ag.segments])
 
-        assert_equal(self._in_box(cen), True)
+        assert_equal(self._in_box(cen, u), True)
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/core/test_atomgroup.py]: assert

File: testsuite/MDAnalysisTests/core/test_atomgroup.py [hunk 4]
```diff
@@ -503,86 +497,87 @@ def test_split_residues(self):
...
-    def test_wrap_residues(self):
-        ag = self.u.atoms[300:400]
+    def test_wrap_residues(self, u):
+        ag = u.atoms[300:400]
         ag.wrap(compound='residues')
 
         cen = np.vstack([r.atoms.center_of_mass() for r in ag.residues])
 
-        assert_equal(self._in_box(cen), True)
+        assert_equal(self._in_box(cen, u), True)
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/core/test_atomgroup.py]: assert

File: testsuite/MDAnalysisTests/core/test_atomgroup.py [hunk 5]
```diff
@@ -503,86 +497,87 @@ def test_split_residues(self):
...
-    def test_wrap_group(self):
-        ag = self.u.atoms[:100]
+    def test_wrap_group(self, u):
+        ag = u.atoms[:100]
         ag.wrap(compound='group')
 
         cen = ag.center_of_mass()
 
-        assert_equal(self._in_box(cen), True)
+        assert_equal(self._in_box(cen, u), True)
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/core/test_atomgroup.py]: assert

File: testsuite/MDAnalysisTests/core/test_atomgroup.py [hunk 6]
```diff
@@ -503,86 +497,87 @@ def test_split_residues(self):
...
         return (coords >= 0.0).all() and (coords <= box).all()
 
-    def test_wrap_atoms(self):
-        ag = self.u.atoms[100:200]
+    def test_wrap_atoms(self, u):
+        ag = u.atoms[100:200]
         ag.wrap(compound='atoms')
 
-        assert_equal(self._in_box(ag.positions), True)
+        assert_equal(self._in_box(ag.positions, u), True)
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/core/test_atomgroup.py]: assert

File: testsuite/MDAnalysisTests/core/test_atomgroup.py [hunk 7]
```diff
@@ -23,19 +23,16 @@
...
 import numpy as np
 import warnings
 
 from numpy.testing import (
-    assert_,
-    assert_array_equal,
     assert_almost_equal,
     assert_equal,
-    assert_array_almost_equal,
+    assert_array_equal,
```
Comment (richardjgowers) [inline testsuite/MDAnalysisTests/core/test_atomgroup.py]: we're not using `assert_array_equal`

Comment (utkbansal) [reply testsuite/MDAnalysisTests/core/test_atomgroup.py]: @richardjgowers The `dtype` causes a mismatch in this case. How do I work around it?

```
E           AssertionError: 
E           Arrays are not almost equal to 7 decimals
E           
E           (mismatch 33.3333333333%)
E            x: array([ 2.6028271, -1.1962211,  0.       ], dtype=float32)
E            y: array([ 2.6028272, -1.1962211,  0.       ])
```

Comment (kain88-de) [reply testsuite/MDAnalysisTests/core/test_atomgroup.py]: which test causes the dtype issue?

Comment (utkbansal) [reply testsuite/MDAnalysisTests/core/test_atomgroup.py]: in class `TestAtomGroup`
```
    def test_index_slice_empty(self, universe):
        u = universe
        assert_equal(u.atoms[0:0], [],
                           "making an empty AtomGroup failed")
```

**Comment by kain88-de** [comment]
@richardjgowers can you have a look at the recent commit to remove `assert_array_*`. Thanks.

**Comment by richardjgowers** [comment]
@kain88-de lgtm if it passes

**Pull request merged**